sites/cheerpj/src/content/docs/13-tutorials/05-interoperability-tutorial.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: Java and JavaScript Interoperability
+description: Learn how to enable Java and JavaScript communication using CheerpJ.
+---
+
+With CheerpJ, a Java application can integrate with JavaScript in a browser environment, enabling seamless interaction between Java and JavaScript code.
+
+In this tutorial, we’ll explore creating a two-way communication channel between Java and JavaScript using CheerpJ. The idea is to allow a Java class to access JavaScript functions and vice versa, to enable sending data between the two environments.
+
+## Prerequisites
+
+{/* - [Download the template project](/docs/cheerpj3/tutorials/CheerpJInteroperabilityTutorial.zip) and unzip it. */}
+
+- [Download the application files](/docs/cheerpj3/tutorials/CheerpJInteroperabilityTutorial.zip) and extract them.
+- [Node.js](https://nodejs.org/en/) (>= 18)
+
+## Project Structure
+
+The project consists of the following files and directories:
+
+```plaintext
+CheerpJInteroperbailityTutorial/
+├── com/example
+│   └── Example.java
+│   └── manifest.txt
+├── index.html
+└── example.jar
+└── Makefile
+```
+
+## Setting Up the Project
+
+We start by creating a simple Java class with a native method that will interact with JavaScript. This class will receive input from JavaScript, process it, and send a response back to JavaScript.
+
```

Comment:
Maybe we could name this section `Breaking down the project files` or similar. We already provide people with all the parts of the code, there is not need for people to write the class from scratch. We could explain we provide the compiled .jar file and we will have a look at the java source code to see what is happening there. 

We could also point out that we provide a Makefile and if people have a local java installation they can try to change some of the alert messages and recompile the jar and test it again.